plugins/database/quota/src/org/apache/cloudstack/quota/QuotaServiceImpl.java

```diff
@@ -0,0 +1,304 @@
+    @Inject
+    private QuotaBalanceDao _quotaBalanceDao;
+    @Inject
+    private QuotaResponseBuilder _respBldr;
+
+    private TimeZone _usageTimezone;
+    private int _aggregationDuration = 0;
+
+    final static BigDecimal s_hoursInMonth = new BigDecimal(30 * 24);
+    final static BigDecimal s_minutesInMonth = new BigDecimal(30 * 24 * 60);
+    final static BigDecimal s_gb = new BigDecimal(1024 * 1024 * 1024);
+
+    public QuotaServiceImpl() {
+        super();
+    }
+
+    @Override
+    public boolean configure(String name, Map<String, Object> params) throws ConfigurationException {
+        super.configure(name, params);
+        String timeZoneStr = _configDao.getValue(Config.UsageAggregationTimezone.toString());
+        String aggregationRange = _configDao.getValue(Config.UsageStatsJobAggregationRange.toString());
+        if (timeZoneStr == null) {
+            timeZoneStr = "GMT";
+        }
+        _usageTimezone = TimeZone.getTimeZone(timeZoneStr);
+
+        _aggregationDuration = Integer.parseInt(aggregationRange);
+        if (_aggregationDuration < UsageUtils.USAGE_AGGREGATION_RANGE_MIN) {
+            s_logger.warn("Usage stats job aggregation range is to small, using the minimum value of " + UsageUtils.USAGE_AGGREGATION_RANGE_MIN);
+            _aggregationDuration = UsageUtils.USAGE_AGGREGATION_RANGE_MIN;
+        }
+        if (s_logger.isDebugEnabled()){
+            s_logger.debug("Usage timezone = " + _usageTimezone + " AggregationDuration=" + _aggregationDuration);
+        }
+        return true;
+    }
+
+    @Override
+    public List<Class<?>> getCommands() {
+        final List<Class<?>> cmdList = new ArrayList<Class<?>>();
+        if (!isQuotaServiceEnabled()) {
+            return cmdList;
+        }
+        cmdList.add(QuotaStatementCmd.class);
+        cmdList.add(QuotaBalanceCmd.class);
+        cmdList.add(QuotaTariffListCmd.class);
+        cmdList.add(QuotaTariffUpdateCmd.class);
+        cmdList.add(QuotaCreditsCmd.class);
+        cmdList.add(QuotaEmailTemplateListCmd.class);
+        cmdList.add(QuotaEmailTemplateUpdateCmd.class);
+        return cmdList;
+    }
+
+    @Override
+    public String getConfigComponentName() {
+        return "QUOTA-PLUGIN";
+    }
+
+    @Override
+    public ConfigKey<?>[] getConfigKeys() {
+        return new ConfigKey<?>[] { QuotaPluginEnabled, QuotaEnableEnforcement, QuotaCurrencySymbol, QuotaSmtpHost, QuotaSmtpPort, QuotaSmtpTimeout, QuotaSmtpUser,
+                QuotaSmtpPassword, QuotaSmtpAuthType, QuotaSmtpSender };
+    }
+
+    public Boolean isQuotaServiceEnabled() {
+        return QuotaPluginEnabled.value();
+    }
+
+    @Override
+    public List<QuotaBalanceVO> findQuotaBalanceVO(Long accountId, String accountName, Long domainId, Date startDate, Date endDate) {
+        final short opendb = TransactionLegacy.currentTxn().getDatabaseId();
+        TransactionLegacy.open(TransactionLegacy.CLOUD_DB).close();
```

Comment:
I am not sure how much that will help in code quality and readability as these DAOs are not part of usage DB that is one reason why we need to switch to cloud db. I think a proxy for such accesses is a better solution as it will consolidate all the access to cloud db in once place.